packages/@icon-magic/distribute/src/index.ts

```diff
@@ -172,69 +168,57 @@ async function distributeByResolution(icon: Icon, outputPath: string) {
   return Promise.all(promises);
 }
 
+function sortIcons(icons: IterableIterator<Icon>) {
+  return Array.from(icons).sort((iconOne: Icon, iconTwo: Icon) => {
+    const iconNameOne = iconOne.iconName;
+    const iconNameTwo = iconTwo.iconName;
+    if (iconNameOne < iconNameTwo) {
+      return -1;
+    }
+    if (iconNameOne > iconNameTwo) {
+      return 1;
+    }
+    return 0;
+  });
+}
+
 /**
- * Creates a sprite and appends SVG icons
- * @param iconSet set of icons to be added to the sprite
- * @param outputPath path to write sprite to
+ *
+ * @param icon icon to distribute
+ * @param outputPath path to move to
  * @param groupByCategory (for sprite creation) whether to group by the category attribute
+ * @param spriteNames object for mapping sprite name to (and storing) svg document and element
+ * @returns promise after completion
  */
-export async function createSprite(
-  iconSet: IconSet,
+async function distributeSvg(
+  icon: Icon,
   outputPath: string,
-  groupByCategory: boolean
+  groupByCategory: boolean,
+  spriteNames: spriteConfig
 ): Promise<void> {
-  const spriteNames = {};
-  const FILE_TYPE = 'svg';
-  let spriteName = 'icons';
-  for (const icon of iconSet.hash.values()) {
-    LOGGER.debug(`adding ${icon.iconName} to ${spriteName}`);
-    if (
-      icon.distribute &&
-      icon.distribute.svg &&
-      icon.distribute.svg.toSprite
-    ) {
-      const spriteAssets = getIconFlavorsByType(icon, FILE_TYPE);
-      const iconSpriteName = icon.distribute.svg.spriteName;
-      spriteName = iconSpriteName ? iconSpriteName : spriteName;
-      let DOCUMENT, svgEl;
-      if (!spriteNames.hasOwnProperty(spriteName)) {
-        ({ DOCUMENT, svgEl } = createSVGDoc());
-        spriteNames[spriteName] = { DOCUMENT, svgEl };
-      } else {
-        ({ DOCUMENT, svgEl } = spriteNames[spriteName]);
-      }
-      for (const asset of spriteAssets) {
-        await appendToSvgDoc(
-          asset,
-          DOCUMENT,
-          svgEl,
-          groupByCategory && icon.category ? icon.category : ''
-        );
-      }
-    }
-  }
-
-  for (spriteName in spriteNames) {
-    const svgEl = spriteNames[spriteName].svgEl;
-    await saveContentToFile(
-      outputPath,
-      spriteName,
-      convertSVGToString(svgEl),
-      FILE_TYPE
+  LOGGER.debug(`distributeSvg for ${icon.iconName}`);
```

Comment:
Be more verbose. It's good to use icon.iconPath's in the logs so you know exactly which icon